### PR TITLE
 chore(core): enabled websocket in route and service

### LINKED ISSUE #1791: change: enabled websocket  in route
I think websocket should be enabled in route, not upstream. This logic is clearer. On the other hand, if we support multiple upstream, we only need to configure it once in the route, otherwise we need to configure each upstream.

### PR DESCRIPTION
**Pull request by dabue**
change: enabled websocket in route

resolve #1791

Pre-submission checklist:
enable websocket
update the test cases
update the user's guide
this is a non-backward compatible PR

### CODE REVIEW

**Comment by membphis** [comment]
@dabue CI failed, please take a look. please confirm you rebased with the latest master branch.

File: t/node/upstream-websocket.t [hunk 1]
```diff
@@ -33,16 +33,15 @@ __DATA__
...
-                            },
-                            "enable_websocket": true,
-                            "type": "roundrobin"
+                [[{
+                    "upstream": {
+                        "nodes": {
+                            "127.0.0.1:1980": 1
                         },
-                        "uri": "/websocket_handshake"
-                }]]
```
Comment (membphis) [inline t/node/upstream-websocket.t:45]: missing `}]]`

Comment (dabue) [reply t/node/upstream-websocket.t:45]: I've fixed it,  please review again :) @membphis

**Review by membphis — APPROVED** [review]

**Comment by membphis** [comment]
@dabue I think we can merge this PR soon ^_^

**Review by spacewander — APPROVED** [review]

**Pull request merged**